static/js/publisher/release/promoteButton.js

```diff
@@ -1,46 +1,70 @@
-import React from "react";
+import React, { Component } from "react";
 import PropTypes from "prop-types";
 
 import ContextualMenu from "./contextualMenu";
 
-export default class PromoteButton extends ContextualMenu {
+export default class PromoteButton extends Component {
+  constructor(props) {
+    super(props);
+    this.setMenuRef = menu => (this.menu = menu);
+  }
+
   promoteToChannelClick(targetChannel, event) {
     this.props.promoteToChannel(targetChannel);
-    this.itemClickHandler(event);
-  }
 
-  renderIcon() {
-    return "⤴";
+    if (this.menu) {
+      this.menu.itemClickHandler(event);
+    }
   }
 
   renderItems() {
-    const { track } = this.props;
-
     return (
       <span className="p-contextual-menu__group">
         <span className="p-contextual-menu__item">Promote to:</span>
-        {this.props.targetRisks.map(targetRisk => {
+        {this.props.targetChannels.map(targetChannel => {
```

Comment:
Could be moved into its own method that contains the logic?

Reply:
Good point, will do.